map_blocks error with dask==2021.4

#### Problem description

Tests fail:

<details>

FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim0-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim0-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim1-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim1-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim2-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim2-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-time-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-time-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim4-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges0-dim4-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim0-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim0-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim1-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim1-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim2-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim2-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-time-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-time-chunked_ds]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim4-chunked_da]
FAILED xskillscore/tests/test_contingency.py::test_Contingency_table[category_edges1-dim4-chunked_ds]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-True-time-observation_1d_long-forecast_1d_long]
~FAILED xskillscore/tests/test_probabilistic.py::test_rps_category_edges_None[True]~
~FAILED xskillscore/tests/test_probabilistic.py::test_rps_category_edges_None[False]~
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-True-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-True-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-False-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-False-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-True-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-False-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-False-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-True-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-False-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-True-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-False-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-True-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-Dataset-True-False-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-True-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-False-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-True-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-False-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-False-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-True-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_tuple-DataArray-True-False-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-True-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-True-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-True-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-False-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-False-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-True-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-False-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-False-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-True-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-False-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-False-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-True-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-False-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-True-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-DataArray-True-False-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[area-Dataset-True-False-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-True-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-False-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-False-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-True-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-True-time-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-False-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-False-None-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-True-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-True-time-observation_3d-forecast_3d]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-Dataset-True-True-None-observation_1d_long-forecast_1d_long]
FAILED xskillscore/tests/test_probabilistic.py::test_roc_returns[all_as_metric_dim-DataArray-True-True-None-observation_1d_long-forecast_1d_long]

</details>

Errors look like:
```bash
skillscore/tests/test_probabilistic.py:831: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
xskillscore/core/probabilistic.py:1209: in roc
    dim=dim,
xskillscore/core/contingency.py:122: in __init__
    self._table = self._get_contingency_table(dim)
xskillscore/core/contingency.py:167: in _get_contingency_table
    bin_dim_suffix="_bin",
xskillscore/core/utils.py:166: in histogram
    return xhist(*args, bins=bins, **kwargs)
/usr/share/miniconda/envs/xskillscore-minimum-tests/lib/python3.7/site-packages/xhistogram/xarray.py:146: in histogram
    block_size=block_size
/usr/share/miniconda/envs/xskillscore-minimum-tests/lib/python3.7/site-packages/xhistogram/core.py:271: in histogram
    block_size=block_size,
/usr/share/miniconda/envs/xskillscore-minimum-tests/lib/python3.7/site-packages/xhistogram/core.py:131: in _histogram_2d_vectorized
    bin_indices = ravel_multi_index(each_bin_indices, hist_shapes)
/usr/share/miniconda/envs/xskillscore-minimum-tests/lib/python3.7/site-packages/xhistogram/duck_array_ops.py:24: in f
    return getattr(module, name)(*args, **kwargs)
<__array_function__ internals>:6: in ravel_multi_index
    ???
/usr/share/miniconda/envs/xskillscore-minimum-tests/lib/python3.7/site-packages/dask/array/core.py:1525: in __array_function__
    return da_func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

multi_index = [dask.array<digitize, shape=(1, 200), dtype=int64, chunksize=(1, 200), chunktype=numpy.ndarray>, dask.array<digitize, shape=(1, 200), dtype=int64, chunksize=(1, 200), chunktype=numpy.ndarray>]
dims = [4, 4], mode = 'raise', order = 'C'

    @wraps(np.ravel_multi_index)
    def ravel_multi_index(multi_index, dims, mode="raise", order="C"):
>       return multi_index.map_blocks(
            _ravel_multi_index_kernel,
            dtype=np.intp,
            chunks=(multi_index.shape[-1],),
            drop_axis=0,
            func_kwargs=dict(dims=dims, mode=mode, order=order),
        )
E       AttributeError: 'list' object has no attribute 'map_blocks'
```

#### Where

I didnt change anything in contingency, only RPS.

https://github.com/xarray-contrib/xskillscore/pull/287/checks?check_run_id=2388515699

versions:
dask                      2021.4.0           pyhd8ed1ab_0    conda-forge
dask-core                 2021.4.0           pyhd8ed1ab_0    conda-forge
xarray                    0.17.0             pyhd8ed1ab_0    conda-forge
xhistogram                0.1.3              pyhd8ed1ab_0    conda-forge
xskillscore               0.0.post1                 dev_0    <develop>


Likely an upstream xhist issue.

--- a/ci/dev.yml
+++ b/ci/dev.yml
@@ -22,7 +22,9 @@ dependencies:
   - scikit-learn
   - scipy
   - xarray>=0.16.1
-  - xhistogram>=0.1.2
+  # xhistogram 0.1.3 introduced an error that broke xskillscore
+  # see https://github.com/xgcm/xhistogram/issues/48
+  - xhistogram!=0.1.3
   # Package Management
   - asv
   - black

--- a/ci/doc.yml
+++ b/ci/doc.yml
@@ -19,6 +19,8 @@ dependencies:
   - sphinx-autosummary-accessors
   - toolz
   - xarray>=0.16.1
-  - xhistogram>=0.1.2
+  # xhistogram 0.1.3 introduced an error that broke xskillscore
+  # see https://github.com/xgcm/xhistogram/issues/48
+  - xhistogram!=0.1.3
   - pip:
       - -e ..

--- a/ci/docs_notebooks.yml
+++ b/ci/docs_notebooks.yml
@@ -12,7 +12,9 @@ dependencies:
   - scikit-learn
   - scipy
   - xarray>=0.16.1
-  - xhistogram>=0.1.2
+  # xhistogram 0.1.3 introduced an error that broke xskillscore
+  # see https://github.com/xgcm/xhistogram/issues/48
+  - xhistogram!=0.1.3
   - importlib_metadata
   - jupyterlab
   - matplotlib-base

--- a/ci/minimum-tests.yml
+++ b/ci/minimum-tests.yml
@@ -12,7 +12,9 @@ dependencies:
   - scikit-learn
   - scipy
   - xarray>=0.16.1
-  - xhistogram>=0.1.2
+  # xhistogram 0.1.3 introduced an error that broke xskillscore
+  # see https://github.com/xgcm/xhistogram/issues/48
+  - xhistogram!=0.1.3
   - coveralls
   - pytest
   - pytest-cov

--- a/requirements.txt
+++ b/requirements.txt
@@ -8,4 +8,6 @@ scikit-learn
 scipy
 toolz
 xarray>=0.16.1
-xhistogram>=0.1.2
+# xhistogram 0.1.3 introduced an error that broke xskillscore
+# see https://github.com/xgcm/xhistogram/issues/48
+xhistogram!=0.1.3

